src/main/java/org/sonarlint/intellij/util/CompoundIcon.java

```diff
@@ -0,0 +1,172 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.util;
+
+import java.awt.Component;
+import java.awt.Graphics;
+import javax.swing.Icon;
+
+public class CompoundIcon implements Icon {
+  public enum Axis {
+    X_AXIS,
+    Y_AXIS,
+    Z_AXIS;
+  }
+
+  public static final float TOP = 0.0f;
+  public static final float LEFT = 0.0f;
+  public static final float CENTER = 0.5f;
+  public static final float BOTTOM = 1.0f;
+  public static final float RIGHT = 1.0f;
+
+  private Icon[] icons;
+
+  private Axis axis;
+
+  private int gap;
+
+  private float alignmentX = CENTER;
+  private float alignmentY = CENTER;
+
+  public CompoundIcon(Icon... icons) {
+    this(Axis.X_AXIS, icons);
+  }
+
+  public CompoundIcon(Axis axis, Icon... icons) {
+    this(axis, 0, icons);
+  }
+
+  public CompoundIcon(Axis axis, int gap, Icon... icons) {
+    this(axis, gap, CENTER, CENTER, icons);
+  }
+
+  public CompoundIcon(Axis axis, int gap, float alignmentX, float alignmentY, Icon... icons) {
+    this.axis = axis;
+    this.gap = gap;
+    this.alignmentX = alignmentX > 1.0f ? 1.0f : (alignmentX < 0.0f ? 0.0f : alignmentX);
+    this.alignmentY = alignmentY > 1.0f ? 1.0f : (alignmentY < 0.0f ? 0.0f : alignmentY);
+
+    for (int i = 0; i < icons.length; i++) {
+      if (icons[i] == null) {
+        String message = "Icon (" + i + ") cannot be null";
+        throw new IllegalArgumentException(message);
+      }
+    }
+
+    this.icons = icons;
+  }
+
+  public Axis getAxis() {
+    return axis;
+  }
+
+  public int getGap() {
+    return gap;
+  }
+
+  public float getAlignmentX() {
+    return alignmentX;
+  }
+
+  public float getAlignmentY() {
+    return alignmentY;
+  }
+
+  public int getIconCount() {
+    return icons.length;
+  }
+
+  public Icon getIcon(int index) {
+    return icons[index];
+  }
+
+  @Override
+  public int getIconWidth() {
+    int width = 0;
+
+    if (axis == Axis.X_AXIS) {
+      width += (icons.length - 1) * gap;
+
+      for (Icon icon : icons) {
+        width += icon.getIconWidth();
+      }
+    } else {
+      for (Icon icon : icons) {
+        width = Math.max(width, icon.getIconWidth());
+      }
+    }
+
+    return width;
+  }
+
+  @Override
+  public int getIconHeight() {
+    int height = 0;
+
+    if (axis == Axis.Y_AXIS) {
+      height += (icons.length - 1) * gap;
+
+      for (Icon icon : icons) {
+        height += icon.getIconHeight();
+      }
+    } else {
+      for (Icon icon : icons) {
+        height = Math.max(height, icon.getIconHeight());
+      }
+    }
+
+    return height;
+  }
+
+  @Override
+  public void paintIcon(Component c, Graphics g, int x, int y) {
+    if (axis == Axis.X_AXIS) {
+      int height = getIconHeight();
+
+      for (Icon icon : icons) {
+        int iconY = getOffset(height, icon.getIconHeight(), alignmentY);
+        icon.paintIcon(c, g, x, y + iconY);
+        x += icon.getIconWidth() + gap;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "x". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1226)